src/WholeBodyStateDisplay.cpp

```diff
@@ -746,9 +752,12 @@ void WholeBodyStateDisplay::processWholeBodyState() {
         boost::shared_ptr<ArrowVisual> arrow;
         arrow.reset(new ArrowVisual(context_->getSceneManager(), scene_node_));
         if (grf_locate_at_cop_ && cop_enable_ && active_contact_in_cop && is_contact_6d) {
-          arrow->setArrow(cop_pos_[i], contact_for_orientation);
+          // Find the rotation between orientation (robot) and contact_orientation (surface), which we need to add on
+          // to contact_for_orientation to ensure the arrow is pointing in the right direction
+          Ogre::Quaternion surface_rotation_adjustment = orientation * contact_orientation.Inverse();
+          arrow->setArrow(cop_pos_[i], surface_rotation_adjustment * contact_for_orientation);
```

Comment:
I had to account for the rotation the robot and the contact surface to get the contact forces and friction cones pointing in the right direction.